src/NServiceBus.Core/Installation/InstallationComponent.cs

```diff
@@ -10,72 +10,117 @@
-            if (!shouldRunInstallers)
+            if (!configuration.ShouldRunInstallers)
             {
-                return;
+                return component;
             }
 
             foreach (var installerType in concreteTypes.Where(t => IsINeedToInstallSomething(t)))
             {
                 containerComponent.ContainerConfiguration.ConfigureComponent(installerType, DependencyLifecycle.InstancePerCall);
             }
+
+            return component;
         }
 
         public async Task Start()
         {
-            if (!shouldRunInstallers)
+            if (!configuration.ShouldRunInstallers)
             {
                 return;
             }
 
-            var queueBindings = settings.Get<QueueBindings>();
-            var username = GetInstallationUserName();
+            var installationUserName = GetInstallationUserName();
 
-            if (settings.CreateQueues())
+            if (configuration.ShouldCreateQueues)
             {
-                await receiveComponent.CreateQueuesIfNecessary(queueBindings, username).ConfigureAwait(false);
+                await receiveComponent.CreateQueuesIfNecessary(queueBindings, installationUserName).ConfigureAwait(false);
             }
 
             foreach (var installer in containerComponent.Builder.BuildAll<INeedToInstallSomething>())
             {
-                await installer.Install(username).ConfigureAwait(false);
+                await installer.Install(installationUserName).ConfigureAwait(false);
             }
         }
 
         string GetInstallationUserName()
         {
-            if (!settings.TryGet("Installers.UserName", out string userName))
+            if (configuration.InstallationUserName != null)
             {
-                if (Environment.OSVersion.Platform == PlatformID.Win32NT)
-                {
-                    userName = $"{Environment.UserDomainName}\\{Environment.UserName}";
-                }
-                else
-                {
-                    userName = Environment.UserName;
-                }
+                return configuration.InstallationUserName;
+            }
+
+            if (Environment.OSVersion.Platform == PlatformID.Win32NT)
+            {
+                return $"{Environment.UserDomainName}\\{Environment.UserName}";
             }
 
-            return userName;
+            return Environment.UserName;
         }
 
-        ReadOnlySettings settings;
+        Configuration configuration;
         ContainerComponent containerComponent;
         ReceiveComponent receiveComponent;
-        bool shouldRunInstallers;
+        QueueBindings queueBindings;
 
         static bool IsINeedToInstallSomething(Type t) => typeof(INeedToInstallSomething).IsAssignableFrom(t);
+
+        public class Configuration
+        {
+            public Configuration(SettingsHolder settings)
+            {
+                this.settings = settings;
+
+                settings.SetDefault("Transport.CreateQueues", true);
```

Comment:
Given that all access should now go over this component configuration is this still necessary or could the getter simply return true if no default is added? Or this is here solely for backward compat?

Reply:
I just did it to make the code simpler we can do that logic in the getter. That said I wonder if this better since if we had plain properties wouldn't we replace this with `ShouldCreateQueues = true;`  ? (Ie put the defaults in the ctor)